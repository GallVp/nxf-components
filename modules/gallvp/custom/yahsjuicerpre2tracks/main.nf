process CUSTOM_YAHSJUICERPRE2TRACKS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/biopython:1.75'
        : 'biocontainers/biopython:1.75'}"

    input:
    tuple val(meta), path(liftover_agp)
    path(scale)

    output:
    tuple val(meta), path("*.assembly"), emit: assembly
    tuple val(meta), path("*.bedpe"), emit: bedpe
    tuple val(meta), path("*.bed"), emit: bed
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    template('yahsjuicerpre2tracks.py')

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """

    touch ${prefix}.assembly
    touch ${prefix}.bedpe
    touch ${prefix}.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
        biopython: \$(pip list | grep "biopython" | cut -d' ' -f3)
    END_VERSIONS
    """
}
