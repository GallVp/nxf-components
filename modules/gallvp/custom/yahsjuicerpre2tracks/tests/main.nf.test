nextflow_process {

    name "Test Process CUSTOM_YAHSJUICERPRE2TRACKS"
    script "../main.nf"
    process "CUSTOM_YAHSJUICERPRE2TRACKS"

    tag "modules"
    tag "modules_gallvp"
    tag "custom"
    tag "custom/yahsjuicerpre2tracks"

    test("inline - agp") {

        when {
            process {
                """
                def test_file = new File('inline.liftover.agp')
                test_file.text = "ctg00000001.1	1	15001	1	W	HYh1_Chr01:6000000-6015000	1	15001	+\\nctg00000002.1	1	15001	1	W	HYh2_Chr01:6000000-6015000	1	15001	+\\n"

                def scale_file = new File('inline.scale')
                scale_file.text = "1\\n"

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    test_file.toPath()
                ]
                input[1] = scale_file.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("inline - agp - scale - 2") {

        when {
            process {
                """
                def test_file = new File('inline.liftover.agp')
                test_file.text = "ctg00000001.1	1	15001	1	W	HYh1_Chr01:6000000-6015000	1	15001	+\\nctg00000002.1	1	15001	1	W	HYh2_Chr01:6000000-6015000	1	15001	+\\n"

                def scale_file = new File('inline.scale')
                scale_file.text = "2\\n"

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    test_file.toPath()
                ]
                input[1] = scale_file.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("inline - agp - stub") {
        options '-stub'

        when {
            process {
                """
                def test_file = new File('inline.liftover.agp')
                test_file.text = "ctg00000001.1	1	15001	1	W	HYh1_Chr01:6000000-6015000	1	15001	+\\nctg00000002.1	1	15001	1	W	HYh2_Chr01:6000000-6015000	1	15001	+\\n"

                def scale_file = new File('inline.scale')
                scale_file.text = "1\\n"

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    test_file.toPath()
                ]
                input[1] = scale_file.toPath()
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}