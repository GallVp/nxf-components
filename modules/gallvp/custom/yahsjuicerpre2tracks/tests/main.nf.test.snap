{
    "inline - agp - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,cab4fe8f578236b699ee2dd5f021ce3c"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bedpe": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cab4fe8f578236b699ee2dd5f021ce3c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-12T15:28:39.683616"
    },
    "inline - agp - scale - 2": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,24dcf28c417a450dc0c10c823ad38c31"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,4533bb10160357ac37dd36c33bb15757"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,1fbc21653249d9f90db49a9a4770e891"
                    ]
                ],
                "3": [
                    "versions.yml:md5,cab4fe8f578236b699ee2dd5f021ce3c"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,24dcf28c417a450dc0c10c823ad38c31"
                    ]
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,1fbc21653249d9f90db49a9a4770e891"
                    ]
                ],
                "bedpe": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,4533bb10160357ac37dd36c33bb15757"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cab4fe8f578236b699ee2dd5f021ce3c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-15T10:09:24.947718"
    },
    "inline - agp": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,72dfdabd22760a91b81ed4241807a0e7"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,a4ec178eaaa2e96b724f927ce9678b5a"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,bc903fc7f717a627853e93ff6c2dfcb3"
                    ]
                ],
                "3": [
                    "versions.yml:md5,cab4fe8f578236b699ee2dd5f021ce3c"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,72dfdabd22760a91b81ed4241807a0e7"
                    ]
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bed:md5,bc903fc7f717a627853e93ff6c2dfcb3"
                    ]
                ],
                "bedpe": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bedpe:md5,a4ec178eaaa2e96b724f927ce9678b5a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,cab4fe8f578236b699ee2dd5f021ce3c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-12T20:38:05.850707"
    }
}